rust/datafusion/src/execution/context.rs

```diff
@@ -62,10 +62,31 @@ use crate::sql::{
 };
 use crate::table::Table;
 
+/// Configuration options for execution context
+#[derive(Copy, Clone)]
+pub struct ExecutionConfig {
+    /// Maximum number of concurrent threads for query execution
+    max_concurrency: usize,
```

Comment:
Isn't it simpler to make this value public instead of having `with_max_concurrency` as a setter?

Reply:
I was following the builder pattern that @houqp introduced with `CsvReadOptions`. Once we have more configs we can chain them together, like this:

```
ExecutionConfig::new()
  .with_concurrency(48)
  .with_foo(...)
  .with_bar(...)
```

We can also add validation logic in these methods. For example, I have now added an assertion in `with_concurrency` to assert that the argument is > 0.

Reply:
Thanks for the explanation. I agree.